dts/bindings/mipi-dbi/mipi-dbi-device.yaml

```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+#
+# Common fields for MIPI-DBI devices
+
+include: [base.yaml, power.yaml]
+
+on-bus: mipi-dbi
+
+properties:
```

Comment:
MIPI DBI display controllers might  also support the Tearing Effect (TE) signal which is then used by host controllers to synchronize frame flushing. Currently, I am introducing some boolean properties, namely `te-enable` and `te-inversion`, in our controller's binding. However, I was thing whether or not these two properties should be added either in `mipi-dbi-device.yaml` or `display-controller.yaml`. Then display drivers such as ili9xxx can be updated and support controlling TE per demand.

Reply:
IE something like `mipi_dbi_set_te(const struct *device, bool enable)`, or something like that to enable controllers to set the TE pin?

Reply:
My initial thought was to just introduce a DT property in `mipi-dbi-device.yaml` which would be exercised by both display and host controllers at driver initialization (in doing so source code related to TE can be compiled out when TE property is not present). Introducing a dedicated API for the host would be useful if there is a meaning for TE to be changed dynamically per frame update (I cannot really think of any such use case).

Reply:
Yeah, we probably don't need an API to enable it dynamically. However, I do think we'd need some kind of API for the display driver. The MIPI controller itself cannot simply wait for TE in `mipi_dbi_write_display`, because `mipi_dbi_write_display` may be called multiple times for one "ram write" command. What about an API like `mipi_dbi_wait_te`? Alternatively, the MIPI DBI controller could support registering a callback for the TE event. I prefer the `mipi_dbi_wait_te` option because it is simpler for controllers to implement- they just check if the API is supported, and if not, proceed to write the frame without waiting for the TE signal.

Reply:
> ...mipi_dbi_write_display may be called multiple times for one "ram write" command.

You are right. I totally forgot this use case. 

> What about an API like mipi_dbi_wait_te...

I am in favor of introducing an API rather than using callback functions that can later be used by display controllers to synchronize frame updates. So, if I understand it correctly, display controllers might first invoke `mipi_dbi_wait_te` to notify host controllers that for the next `mipi_dbi_write_display` call, they should first wait for the TE signal to arrive before flushing the frame buffer, correct? Then, subsequent calls to `mipi_dbi_write_display` for the same "ram write" can follow, if needed.  Let me know if my understanding is correct.

Reply:
I was thinking something more like this:

```
    mipi_desc.width = desc->width;
	mipi_desc.height = desc->height;
	/* Per MIPI API, pitch must always match width */
	mipi_desc.pitch = desc->width;

    /* Wait for display to send TE signal */
    mipi_dbi_wait_te(dev);

    /* Send RAM write command */
	r = ili9xxx_transmit(dev, ILI9XXX_RAMWR, NULL, 0);
	if (r < 0) {
		return r;
	}

	for (write_cnt = 0U; write_cnt < nbr_of_writes; ++write_cnt) {
        /* Send display data */
		r = mipi_dbi_write_display(config->mipi_dev,
					   &config->dbi_config,
					   write_data_start,
					   &mipi_desc,
					   data->pixel_format);
```

That being said, the TE signal is usually a GPIO. Is there any reason your controller requires that TE signal to be routed to the peripheral, versus to a GPIO that the display driver can wait for a signal from?

Reply:
Understood. Well, our host controller supports asserting its interrupt line based on the programmed TE polarity and triggering mode (level or edge) and this is why the TE signal is routed to the host controller rather than being handled as a regular GPIO. I do understand that this feature might not be present in other host controllers and so TE handling should be done by display drivers, treating it as a regular GPIO.

Reply:
Sure, the introduction of `mipi_dbi_wait_te(config->mipi_dev)` is what is needed. Let's handle this in a separate PR. The shield I have at my disposal breaks out the TE signal for the ILI9341 so I can test with TE. I am currently OoO so I can take care of this once I am back at work.